python/cuml/linear_model/logistic_regression_mg.pyx

```diff
@@ -0,0 +1,198 @@
+        int linesearch_max_iter
+        int lbfgs_memory
+        int verbose
+        bool fit_intercept
+        bool penalty_normalized
+
+
+cdef extern from "cuml/linear_model/qn_mg.hpp" namespace "ML::GLM::opg" nogil:
+
+    void qnFit(
+        const handle_t& handle,
+        const qn_params& pams, 
+        float *X,
+        bool X_col_major,
+        float *y,
+        int N,
+        int D,
+        int C,
+        float *w0,
+        float *f,
+        int *num_iters,
+        int n_samples,
+        int rank,
+        int n_ranks) except +
+
+
+class LogisticRegressionMG(LogisticRegression):
+    
+    def __init__(self, *, handle=None):
+        super().__init__(handle=handle)
+
+    def prepare_for_fit(self, n_classes):
+        self.qnparams = QNParams(
+            loss=self.loss,
+            penalty_l1=self.l1_strength,
+            penalty_l2=self.l2_strength,
+            grad_tol=self.tol,
+            change_tol=self.delta
+            if self.delta is not None else (self.tol * 0.01),
+            max_iter=self.max_iter,
+            linesearch_max_iter=self.linesearch_max_iter,
+            lbfgs_memory=self.lbfgs_memory,
+            verbose=self.verbose,
+            fit_intercept=self.fit_intercept,
+            penalty_normalized=self.penalty_normalized
+        )
+
+        # modified
+        qnpams = self.qnparams.params
+
+        # modified qnp
+        solves_classification = qnpams['loss'] in {
+            qn_loss_type.QN_LOSS_LOGISTIC,
+            qn_loss_type.QN_LOSS_SOFTMAX,
+            qn_loss_type.QN_LOSS_SVC_L1,
+            qn_loss_type.QN_LOSS_SVC_L2
+        }
+        solves_multiclass = qnpams['loss'] in {
+            qn_loss_type.QN_LOSS_SOFTMAX
+        }
+
+        if solves_classification:
+            self._num_classes = n_classes
+        else:
+            self._num_classes = 1
+
+        if not solves_multiclass and self._num_classes > 2:
+            raise ValueError(
+                f"The selected solver ({self.loss}) does not support"
+                f" more than 2 classes ({self._num_classes} discovered).")
+
+        if qnpams['loss'] == qn_loss_type.QN_LOSS_SOFTMAX \
+           and self._num_classes <= 2:
+            raise ValueError("Two classes or less cannot be trained"
+                             "with softmax (multinomial).")
+
+        if solves_classification and not solves_multiclass:
+            self._num_classes_dim = self._num_classes - 1
+        else:
+            self._num_classes_dim = self._num_classes
+
+        if self.fit_intercept:
+            coef_size = (self.n_cols + 1, self._num_classes_dim)
+        else:
+            coef_size = (self.n_cols, self._num_classes_dim)
+
+        if self.coef_ is None or not self.warm_start:
+            self.solver_model._coef_ = CumlArray.zeros(
+                coef_size, dtype=self.dtype, order='C')
+
+    def fit(self, X, y, rank, n_ranks, n_samples, n_classes, convert_dtype=False) -> "LogisticRegressionMG":
```

Comment:
As mentioned above, I think this could be simplified further to look more like [this](https://github.com/rapidsai/cuml/blob/00bdad3cae107ce0021739dc11860e9629af738f/python/cuml/linear_model/linear_regression_mg.pyx#L74).

Reply:
Revised the code and the _fit now looks exactly the same. 
fit looks the same except for one additional argument (i.e. order='F'). This is because `LinearRegressionMG` uses 'F' order but `cuml.LogisticRegression` uses 'C' order.